networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -160,13 +160,11 @@ Graph GraphBuilder::toGraph(bool autoCompleteEdges, bool parallel) {
     } else {
         toGraphDirectSwap(G);
     }
-
-    assert(G.outEdges.size() == n);
-    assert(G.outEdgeWeights.size() == (weighted ? n : 0));
-    assert(G.inEdges.size() == (directed ? n : 0));
-    assert(G.inEdgeWeights.size() == ((weighted && directed) ? n : 0));
-
-    G.m = numberOfEdges(G);
+    #ifdef NETWORKIT_SANITY_CHECKS
+    assert(n == G.upperNodeIdBound());
+    assert(G.checkConsistency());
+    #endif
+    G.setEdgeCount(unsafe, numberOfEdges(G));
```

Comment:
`checkConsistency` should be adjusted to also check the edge count. (And hence `setEdgeCount()` should be called before `checkConsistency()`.)